aniwash/src/main/java/aniwash/viewmodels/MainViewModel.java

```diff
@@ -100,18 +103,24 @@ private void addEntriesToCalendar() {
         for (Appointment appointment : appointmentList) {
             Product mainProduct = getMainProduct(appointment.getMainProductId(), appointment.getProductList());
             assert mainProduct != null;
-            Entry<Appointment> entry = new Entry<>(mainProduct.getName("en"), new Interval(appointment.getStartDate(), appointment.getEndDate()), "id" + appointment.getId());
+            if (mainProduct == null) {
+                System.out.println("mainProduct is null");
+                continue;
+            }
+            Entry<Appointment> entry = new Entry<>(mainProduct.getName("en"),
+                    new Interval(appointment.getStartDate(), appointment.getEndDate()), "id" + appointment.getId());
             entry.setLocation(appointment.getAnimalList().get(0).getName());
             entry.setCalendar(calendarMap.get(mainProduct.getName("en")));
             entry.setUserObject(appointment);
         }
     }
 
-    public Appointment createAppointment(ZonedDateTime zdtStart, ZonedDateTime zdtEnd, Customer selectedCustomer, Animal animal, long mainProductId, Map<Product, Discount> p) {
+    public Appointment createAppointment(ZonedDateTime zdtStart, ZonedDateTime zdtEnd, Customer selectedCustomer,
+            Animal animal, long mainProductId, Map<Product, Discount> p, TextArea descriptionText) {
         IAppointmentDao appointmentDao = (AppointmentDao) daoMap.get("appointment");
         Appointment appointment = new Appointment(zdtStart, zdtEnd);
-        appointment.setMainProductId(mainProductId);
-        LocalizedAppointment localAppointment = new LocalizedAppointment(appointment, "Appointment for " + selectedCustomer.getName());
+        LocalizedAppointment localAppointment = new LocalizedAppointment(appointment,
```

Comment:
You can add this description text in LocalizedAppointment(Appointment, Description)